### PR TITLE
[NCL-8166] Implement Product Version create/edit page

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

File: src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+  const submitCreate = (data: IFieldValues) => {
+    return serviceContainerCreatePage
+      .run({
+        serviceData: { data: { ...data, product: { id: productId } } },
+      })
+      .then((response: any) => {
+        const newProductVersionId = response?.data?.id;
+        if (!newProductVersionId) {
+          throw new Error(`Invalid newProductVersionId coming from Orch POST response: ${newProductVersionId}`);
```
Comment (DnsZhou) [inline src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx:77]: Do we need to consider using `uiLogger` as this is a pnc side error that we need to take care of?

File: src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        }
+        navigate(`../${newProductVersionId}`);
+      })
+      .catch((e: any) => {
+        throw new Error('Failed to create Product Version.');
```
Comment (DnsZhou) [inline src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx:82]: same with error above, consider `uiLogger`

File: src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+      attributes: { BREW_TAG_PREFIX: data['attributes.brewTagPrefix'] },
+    });
+
+    serviceContainerEditPagePatch
+      .run({ serviceData: { id: productVersionId, patchData } })
+      .then(() => {
+        navigate(`..`);
+      })
+      .catch(() => {
+        throw new Error('Failed to edit Product Version.');
```
Comment (DnsZhou) [inline src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx:98]: same with error above, consider `uiLogger`

**Review by DnsZhou — APPROVED** [review]
Verified from my local env, code looks good to me except the error that we may consider using `uiLogger`, but feel free to proceed if you think that is not necessary. Thanks!

File: src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+import { PageLayout } from 'components/PageLayout/PageLayout';
+import { ServiceContainerCreatingUpdating } from 'components/ServiceContainers/ServiceContainerCreatingUpdating';
+
+import * as productVersionApi from 'services/productVersionApi';
+
+import { validateProductVersionName } from 'utils/formValidationHelpers';
+import { createSafePatch } from 'utils/patchHelper';
+import { generatePageTitle } from 'utils/titleHelper';
+
+const formConfig = {
```
Comment (matedo1) [inline src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx:22]: Just a reminder we started to use updated name with type

File: src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx (hunk 1 above)
Comment (patrikk0123) [reply src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx:77]: This issue will be considered in different Jira (NCL-8169).

File: src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx (hunk 2 above)
Comment (patrikk0123) [reply src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx:82]: This issue will be considered in different Jira (NCL-8169).

File: src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx (hunk 3 above)
Comment (patrikk0123) [reply src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx:98]: This issue will be considered in different Jira (NCL-8169).

File: src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx (hunk 4 above)
Comment (patrikk0123) [reply src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx:22]: I will rebase the PR and update accordingly.

**Pull request merged**